service/controller/v14/resource/service/master_service.go

```diff
@@ -15,7 +15,8 @@ func newMasterService(customObject v1alpha1.KVMConfig) *apiv1.Service {
 			APIVersion: "v1",
 		},
 		ObjectMeta: apismetav1.ObjectMeta{
-			Name: key.MasterID,
+			Name:      key.MasterID,
+			Namespace: key.ClusterID(customObject),
```

Comment:
for some reason, ns was missing here all the time

Reply:
What happens when we upgrade clusters now? Will there be leftovers in the old ns that was used?

Reply:
It wasn't really a big problem as  when you call to create a function you specify the ns int he calls
```
k8s.CoreV1().Services(namespace).Create(service)
```
but when I was comparing the current and desired service the NS was missing in the desired resource.

TL;DR: no leftovers, all will be same.

Reply:
That's good then :+1: